syz-manager/manager.go

```diff
@@ -686,11 +675,23 @@ func loadProg(target *prog.Target, data []byte) (*prog.Prog, error) {
 }
 
 func (mgr *Manager) fuzzerInstance(ctx context.Context, inst *vm.Instance, updInfo dispatcher.UpdateInfo) {
+	mgr.mu.Lock()
```

Comment:
> We don't reset it to nil under mu, do we?

We do. In `MachineCheck()`.

Reply:
OK, then what happens if it's reset right after unlock?

Reply:
A nil-deref panic here:
https://github.com/google/syzkaller/blob/109d2082129b092aabc13316527c1c35a9c6be1b/syz-manager/manager.go#L743

Reply:
If load/update of mgr.serv are racing, then it's possible, right?

Reply:
Yes

Reply:
So what happens with this race? Can we get a nil deref?

Reply:
Yes, I did observe a nil deref at that exact line. And yes, it's because of a data race on `mgr.serv`. We set `mgr.serv = nil` in `MachineChecked()` and we use it when we call `mgr.serv.StopFuzzing`.